int32 upper_hue
int32 lower_hue
int32 lower_sat
int32 upper_sat
int32 lower_val
int32 upper_val

